libs/muddle/src/kademlia/peer_tracker.cpp

```diff
@@ -753,50 +790,60 @@ void PeerTracker::Periodically()
     }
   }
 
-  if (tracker_configuration_.allow_desired_connections)
+  if (config.allow_desired_connections)
   {
     // Making connections to user defined endpoints
     ConnectToDesiredPeers();
   }
+
   // Ensuring that we keep connections open which we are currently
   // pulling data from
-  for (auto const &item : uri_resolution_tasks_)
   {
-    if (item.first != own_address_)
+    FETCH_LOCK(uri_mutex_);
+    FETCH_LOCK(direct_mutex_);
```

Comment:
We should introduce a form to lock several mutexes all at once in natural order, same as `std::scoped_lock` does.